plasma_framework/contracts/src/framework/ExitGameController.sol

```diff
@@ -8,14 +8,21 @@ import "./utils/ExitPriority.sol";
 import "../utils/TxPosLib.sol";
 
 contract ExitGameController is ExitGameRegistry {
+
     mapping (uint256 => IExitProcessor) public delegations;
-    mapping (address => PriorityQueue) public exitsQueues;
+    mapping (uint256 => bool) public registeredVaults;
```

Comment:
Instead of having two stuffs: vault, token with two sets of actions, I would merge into one and have the following functions/events instead:

```
event ExitQueueAdded(vaultId, token)
addExitQueue(vaultId, token)
hasExitQueue(vaultId, token)
```

and wouldn't need this extra mapping anymore.

I think this is much more simpler and easier to understand.

Reply:
I did so a vault can only be registered by operator. That's why there is registeredVaults map. Do you think that's not an issue that users will be allowed to register vaults and we will get some bogus vaultIds?

Reply:
hmmm, I don't *think* it should be worry-some as we are not limiting `addToken` to operator too. Seems to me who create the queue is not important, and also whether that queue is of some non even existing vault id does not matter.

But let get another pair of eye on this: @paulperegud @pdobacz @kevsul

Reply:
What would be the consequence of someone `addExitQueue` for an important token for all `vaultIds` (is that even possible? is the `vaultID` for new vault chosen by the operator or predetermined (can't add vault 13 if there's no vault 12 yet))

If the above is a non-issue then I think we're cool.

If it is an issue then that would require moving the `PriorityQueues` to be anchored in vaults rather than `PlasmaFramework` which sounds like a uge change. (that is one does `IVault.addToken(token)` instead)

Reply:
@pdobacz Vault id is chosen by operator when registering a vault in VaultRegistry. Right now we allow anyone to create a queue for a token, but vaults can only be registered by operator. My idea was to mirror that behavior.

Reply:
> What would be the consequence of someone addExitQueue for an important token for all vaultIds

My claim is that it would be fine and doesn't matter as long as our ExitGame contracts code it *correctly* when putting things into exit queue.

> is that even possible?

In my proposal it is. It basically allow anyone to generate any arbitrary exit queue.

Reply:
ok, so if the vault id is at the discretion of the operator when they're registering the vault, then I think the above scenario is a non-issue :+1:

Reply:
ok

Reply:
done